### ISSUE TITLE
html5 video slider

### ISSUE THREAD

**Issue by kukac7** [metafizzy/flickity#71]
Hi,

Does not work well with the slider cell with html5 video content.

look: http://codepen.io/kukac7/pen/ogMpNB

What could be the problem?

Thank you!

**Comment by desandro** [comment]
Adding proper support and documentation for HTML5 video is something I still need to do.

But I may have found a short-term solution. See demo CodePen: http://codepen.io/desandro/pen/PwBEgL or vanilla JS CodePen: http://codepen.io/desandro/pen/MYBQbp

``` js
$(document).ready( function() {
  var $gallery = $('.gallery').flickity();

  function onLoadeddata( event ) {
    var cell = $gallery.flickity( 'getParentCell', event.target );
    $gallery.flickity( 'cellSizeChange', cell && cell.element );
  }
  
  $gallery.find('video').each( function( i, video ) {
    video.play();
    $( video ).on( 'loadeddata', onLoadeddata );
  });

});
```

There are two issues to resolve

1. imagesLoaded does not work with video. This demo adds an event listener to resize cells once its size is set, similar to imagesLoaded.

2. [WebKit has a bug where changing the DOM stops autoplay](http://dropshado.ws/post/112695083527/html5-video-autoplay-dom-change-webkit-bug). This demo adds `.play()` to resume autoplay.

**Comment by kukac7** [comment]
thank you! :dancers:

**Comment by dev4apart** [comment]
Hi,

I am also interested and excited to see the support of html5 videos :)

**Comment by dcinzona** [comment]
+1

**Comment by christinechaooo** [comment]
+1! Would be nice to mix video and image.

**Comment by bbnnt** [comment]
as @desandro says, it it supporte (with some fix)
however, the codepens provided do not show any videos on my safari browser.

I can make it work though the console when calling play() on the video, nothing to do doing it from code

**Comment by npredey** [comment]
Is there a way to autoplay the carousel when the video is finished playing? I tried setting the `autoplay` variable to the duration of the video, but nothing moves when the video is finished playing. In fact, even if I set a static `autoplay`, it still does not auto play.

**Comment by desandro** [comment]
@npredey I imagine you want trigger [`next`](http://flickity.metafizzy.co/api.html#next) then [`playPlayer`](http://flickity.metafizzy.co/api.html#playplayer)

**Comment by coolwebs** [comment]
yessss +1

**Comment by stefanedberg** [comment]
+1

**Comment by bryanchalker** [comment]
+1....million

**Comment by dejardine** [comment]
+1

**Comment by HandHugs** [comment]
is this working for anyone? Im trying to get html5 video to play when the slide is in view, pause when user slides to new slide. 

The demos above are not working!

**Comment by tdcpepsi** [comment]
**I have video categories in my website...**
How would I add the the categories to the Flickity slider??? A category such as "travel-and-event". How would I add or pull the video element into the slider, what functions would I use???

**This is an example category:** http://www.vtubemobi.com/cat/travel-and-event

**Comment by brigleb** [comment]
Indeed, I can't get those demos to work, and was really hoping to use Flickity on a current project.

**Comment by ianhobbs** [comment]
yes same here.... to bad

**Comment by jmharveys** [comment]
So there's no way to html 5 video slider with flickity? I made one but when you slide to the next one, the video start.... any idea?!

**Comment by jmharveys** [comment]
??

**Comment by HandHugs** [comment]
I got it working eventually with this

`this.$slideshow.on( 'cellSelect.flickity', function() {
       $('.slideshow-slide').find('video').get(0).pause();
        $('.is-selected').find('video').get(0).play();
       
      });`

**Comment by salunet** [comment]
`var playWholeVideo = true;

					$(this).flickity({
						prevNextButtons: false,
						pageDots: false,
						cellSelector: '.coverItem',
						draggable: false,
						freeScroll: false,
						wrapAround: true,
						autoPlay: false,
						pauseAutoPlayOnHover: false,
						imagesLoaded: true,
						on: {
							select: function (index) {
								var flkty = this;
								$(flkty.selectedElement).find('video').each(function (i, video) {
									if (playWholeVideo) {
										flkty.stopPlayer();
										video.loop = false;
										video.onended = function (e, i) {
											flkty.next();
										};
									}
									video.play();
								});
							},
							change: function (index) {
								if (!playWholeVideo) {
									var flkty = this;
									var previousIndex = index === 0 ? flkty.slides.length - 1 : index - 1;
									var previousElement = flkty.slides[previousIndex].getCellElements();
									$(previousElement).find('video').each(function (i, video) {
										video.pause();
									});
								}
							}
						}`

**Comment by riddy92** [comment]
Could this work with masonry? and if yes, how would it.

**Comment by pommidorka** [comment]
> Could this work with masonry? and if yes, how would it.

Yeap, easy:

```
// init Masonry
var $grid = $('.grid').masonry({
  // options...
});

//call when element loaded
function onLoaded( event ) {
  $grid.masonry('layout')
}

// layout Masonry after each image loads
$grid.imagesLoaded().progress(onLoaded);

//*****//-VIDEO LOAD-//*****//

// layout Masonry after each video loads
$grid.find('video').each( function( i, video ) {
  $( video ).on( 'loadeddata', onLoaded );
});

//*****//-//*****//-//*****//

```
